client/src/document/editthispage.tsx

```diff
@@ -33,34 +34,37 @@ export function EditThisPage({ source }) {
   if (!source) {
     return null;
   }
-  if (source.github_url) {
-    return (
-      <p className="edit-this-page">
-        <a
-          href={source.github_url}
-          title={`Folder: ${source.folder}`}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Edit this page in <b>GitHub</b>
-        </a>
-      </p>
-    );
-  } else if (source.content_file) {
-    return (
-      <p className="edit-this-page">
-        <a
-          href={`file://${source.content_file}`}
-          title={`Folder: ${source.folder}`}
-          onClick={openInEditorHandler}
-        >
-          Edit this page in your editor
-        </a>
-        <br />
-        {opening && <small>Trying to your editor now...</small>}
-      </p>
-    );
-  } else {
-    throw new Error("source has neither .github_url or .content_file");
-  }
+  const url = new URL(source.github_url);
+  const { pathname } = url;
+  const branch = pathname.split("/")[4];
```

Comment:
Branches can have `/` in them. E.g. `dependabot/reactlib1.2.3` but I think this'd be turned into `dependabot%2Freactlib1.2.3` by the time it reaches here.

Reply:
It doesn't do the right path for me, I get: `https://github.com/mdn/stumptown-renderer/blob/peterbe/557-editthispage-should-be-smarter/content/files/en-us`
Which neither opens in GitHub nor in my editor.

Also wondering whether we could just inject the folder into the source object as well, to do less slash-wrangling

Reply:
Oh that's what we did before. Still I think I like it more.

Reply:
> It doesn't do the right path for me, I get: `https://github.com/mdn/stumptown-renderer/blob/peterbe/557-editthispage-should-be-smarter/content/files/en-us`
> Which neither opens in GitHub nor in my editor.

Well, the first problem is that you're inside a branch that came from a different fork. So how is it supposed to know that the link should be `https://github.com/PETERBE/yari/blob...`? 
I can imagine some hacks to figure out which remote a branch came from but none of them feel solid. So for now, I'd be happy that all links start with `https://github.com/mdn/yari/...`.

Reply:
> Oh that's what we did before. Still I think I like it more.

Except it's strangely tricky. 
When a document is Node rendered you have access to `source.filepath` and `folder`. These values look like this for example:
```
SOURCE.filepath /Users/peterbe/dev/MOZILLA/MDN/stumptown-renderer/content/files 
FOLDER /Users/peterbe/dev/MOZILLA/MDN/stumptown-renderer/content/files/en-us/web/html/element/noscript
```

If you calculate the relative of them you get `en-us/web/html/element/noscript` which is lacking in source. So how do you know where it came from if you ultimately don't want to have `/Users/peterbe/...` to appear in the rendered HTML files? You can't. You can perhaps assume that the default is `content/files` as a prefix, which is what the server does, but that's no different from extracting it from the GitHub URL.

Reply:
Can you debug why it's not working in your `server`? Obviously the GItHub URL won't work yet so you'd need to use your imagination. But why isn't your IDE opening it? Can you find out if an error is happening? Also, I don't expect it to be easy on Windows (does it even have a `$EDITOR` concept?) but whatever the situation we should at least have better error handling and debugging on it.